doc/doxygen/src/driver-guide.md

```diff
@@ -0,0 +1,381 @@
+    uint8_t addr;       /**< the device's address on the bus */
+} tmpabc_params_t;
+@endcode
+
+**NOTE:** In many cases it makes sense, to copy the `xxx_params` data into the
+device descriptor during initialization. In some cases, it is however better to
+just link the `params` data via pointer and only copy selected data. This way,
+configuration data that is only used once can be read directly from ROM, while
+often used fields (e.g. used peripherals) are stored directly in the device
+descriptor and one saves hereby one de-referencing step when accessing them.
+
+## Default device configuration                          {#driver-guide-config}
+
+Each device driver in RIOT MUST supply a default configuration file, named
+`DEVNAME_params.h`. This file should be located in the `RIOT/drivers/...`. The
+idea is, that this file can be overridden by an application or a board, by
+simply putting a file with the same name in the application's or the board's
+include folders, while RIOT's build system takes care of preferring those files
+instead of the default params file.
+
+A default parameter header file for the example temperature sensor above would
+look like this (`tmpabc_params.h`):
+
+@code{.c}
+/* ... */
+
+#include "board.h"  /* THIS INCLUDE IS MANDATORY */
+#include "tmpabc.h"
+
+/* ... */
+
+/**
+ * @brief   Default configuration parameters for TMPABC sensors
+ * @{
+ */
+#ifndef TMPABC_PARAM_I2C
+#define TMPABC_PARAM_I2C        (I2C_DEV(0))
+#endif
+#ifndef TMPABC_PARAM_ADDR
+#define TMPABC_PARAM_ADDR       (0xab)
+#endif
+
+#ifndef TMPABC_PARAMS
+#define TMPABC_PARAMS           { .i2c  = TMPABC_PARAM_I2C \
+                                  .addr = TMPABC_PARAM_ADDR }
+#endif
+/** @} */
+
+/**
+ * @brief   Allocation of TMPABC configuration
+ */
+static const tmpabc_params_t tmpabc_params[] = {
+    TMPABC_PARAMS
+}
+/* ... */
+@endcode
+
+Now to influence the default configuration parameters, we have these options:
+
+First, we can override one or more of the parameter from the makesystem, e.g.
+
+@code{.sh}
+CFLAGS="-DTMPABC_PARAM_ADDR=0x23" make all
+@endcode
+
+Second, we can override selected parameters from the board configuration
+(`board.h`):
+
+@code.{c}
+/* ... */
+/**
+ * @brief   TMPABC sensor configuration
+ * @{
+ */
+#define TMPABC_PARAM_I2C        (I2C_DEV(1))
+#define TMPABC_PARAM_ADDR       (0x17)
+/** @} */
+/* ... */
+@endcode
+
+Third, we can define more than a single device in the board configuration
+(`board.h`):
+
+@code{.c}
+/* ... */
+/**
+ * @brief   Configure the on-board temperature sensors
+ * @{
+ */
+#define TMPABC_PARAMS           { \
+                                  .i2c  = I2C_DEV(1), \
+                                  .addr = 0x31 \
+                                }, \
+                                { \
+                                    .i2c  = I2C_DEV(1), \
+                                    .addr = 0x21 \
+                                }
+/** @} */
+/* ... */
+@encode
```

Comment:
> Using `@code` and `@endcode` instead of the fences works reliable on the old version of Doxygen. But that still doesn't work in conjunction with the numbered list.

But it has some other pitfalls ;-)

```suggestion
@endcode
```

Reply:
Done. Thanks!